plugins/modules/route53.py

```diff
@@ -360,136 +347,78 @@
       - 0 issue "ca.example.net"
       - 0 issuewild ";"
       - 0 iodef "mailto:security@example.com"
-
 '''
 
-import time
-import distutils.version
 
 try:
-    import boto
-    import boto.ec2
-    from boto.route53 import Route53Connection
-    from boto.route53.record import Record, ResourceRecordSets
-    from boto.route53.status import Status
+    import botocore
 except ImportError:
-    pass  # Handled by HAS_BOTO
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils._text import to_native
 
 from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import get_aws_connection_info
-from ansible_collections.amazon.aws.plugins.module_utils.ec2 import HAS_BOTO
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_message
+from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
+
+from operator import itemgetter
+
+MAX_AWS_RETRIES = 10  # How many retries to perform when an API call is failing
+WAIT_RETRY = 5  # how many seconds to wait between propagation status polls
+
+
+def remove_none_values_from_dict(dictionary):
+    return dict(filter(lambda i: i[1] is not None, dictionary.items()))
 
 
-MINIMUM_BOTO_VERSION = '2.28.0'
-WAIT_RETRY_SLEEP = 5  # how many seconds to wait between propagation status polls
+@AWSRetry.jittered_backoff(retries=MAX_AWS_RETRIES)
+def get_record(route53, zone_id, record_name, record_type, record_identifier):
+    paginator = route53.get_paginator('list_resource_record_sets')
+    record_sets_results = paginator.paginate(HostedZoneId=zone_id).build_full_result()['ResourceRecordSets']
 
```

Comment:
build_full_result can be a relatively expensive.  It's better to wrap just the paginated call in a dedicate function.  For example:
```
def _list_record_sets(route53, **kwargs):
    paginator = route53.get_paginator('list_resource_record_sets')
    return = paginator.paginate(**kwargs).build_full_result()['ResourceRecordSets']

def get_record(route53, zone_id, record_name, record_type, record_identifier):
    record_set_results = _list_record_sets(HostedZoneId=zone_id)
    ...
```

The value of doing this shows up a little more with get_zone_id_by_name where you're also doing `route53.get_hosted_zone(...)`  where you could let the retry decorator retry just that single call (with aws_retry=True) instead of needing to retry the paginated call *and* the get_hosted_zone call.